addons/stock/wizard/stock_replenishment_info.py

```diff
@@ -78,3 +89,51 @@ def _compute_json_replenishment_history(self):
                 'template': 'stock.replenishmentHistory',
                 'replenishment_history': replenishment_history
             })
+
+
+class StockReplenishmentOption(models.TransientModel):
+    _name = 'stock.replenishment.option'
+    _description = 'Stock warehouse replenishment option'
+
+    route_id = fields.Many2one('stock.route')
+    product_id = fields.Many2one('product.product')
+    replenishment_info_id = fields.Many2one('stock.replenishment.info')
+
+    location_id = fields.Many2one('stock.location', related='warehouse_id.lot_stock_id')
+    warehouse_id = fields.Many2one('stock.warehouse', related='route_id.supplier_wh_id')
+    uom = fields.Char(related='product_id.uom_name')
+    qty_to_order = fields.Float(related='replenishment_info_id.qty_to_order')
+
+    free_qty = fields.Float(compute='_compute_free_qty')
+    lead_time = fields.Char(compute='_compute_lead_time')
+
+    @api.depends('product_id', 'route_id')
+    def _compute_free_qty(self):
+        for record in self:
+            record.free_qty = record.product_id.with_context(location=record.location_id.id).free_qty
```

Comment:
I would use warehouse in context instead of location_id

Reply:
Or does location take sublocation in the quantities? If yes, you could keep it

Reply:
On the free qty field on product.product it states:
_In a context with a single Stock Location, this includes goods stored in this location, or any of its children._

So it can stay like this using the location_id